### PR TITLE
Kiwi-1180: Create Layer 2 stack for the BAV CRI DynamoDB

### PR DESCRIPTION
**Pull request by madankaruppiah**
- Create layer 2 stack for the DynamoDB

- [KIWI-1182](https://govukverify.atlassian.net/browse/KIWI-1182)

[KIWI-1182]: https://govukverify.atlassian.net/browse/KIWI-1182?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infra-l2-dynamo/template.yaml [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+          AttributeType: "S"
+        - AttributeName: "yotiSessionId"
+          AttributeType: "S"
+        - AttributeName: "authSessionState"
+          AttributeType: "S"
+      KeySchema:
+        - AttributeName: "sessionId"
+          KeyType: "HASH"
+      GlobalSecondaryIndexes:
+        - IndexName: "authCode-updated-index"
```
Comment (ViveakV) [inline infra-l2-dynamo/template.yaml:81]: Can get rid of this

File: infra-l2-dynamo/template.yaml [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+          Projection:
+            NonKeyAttributes:
+              - "sessionId"
+              - "redirectUri"
+              - "clientId"
+              - "authSessionState"
+              - "clientSessionId"
+              - "expiryDate"
+            ProjectionType: "INCLUDE"
+        - IndexName: "yotiSessionId-index"
```
Comment (ViveakV) [inline infra-l2-dynamo/template.yaml:94]: Can get rid of this

File: infra-l2-dynamo/template.yaml [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            NonKeyAttributes:
+              - "sessionId"
+              - "subject"
+              - "clientIpAddress"
+              - "state"
+              - "clientId"
+              - "clientSessionId"
+              - "authSessionState"
+            ProjectionType: "INCLUDE"
+        - IndexName: "authSessionState-index"
```
Comment (ViveakV) [inline infra-l2-dynamo/template.yaml:108]: Can get rid of this

File: infra-l2-dynamo/template.yaml (hunk 1 above)
Comment (madankaruppiah) [reply infra-l2-dynamo/template.yaml]: Deleted

File: infra-l2-dynamo/template.yaml [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        - AttributeName: "authorizationCode"
+          AttributeType: "S"
+        - AttributeName: "yotiSessionId"
+          AttributeType: "S"
+        - AttributeName: "authSessionState"
+          AttributeType: "S"
+      KeySchema:
+        - AttributeName: "sessionId"
+          KeyType: "HASH"
+      GlobalSecondaryIndexes:
```
Comment (ViveakV) [inline infra-l2-dynamo/template.yaml:80]: Do we still need this?

File: infra-l2-dynamo/template.yaml [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    Type: "AWS::DynamoDB::Table"
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+    Properties:
+      TableName: !Sub "session-${AWS::StackName}"
+      BillingMode: "PAY_PER_REQUEST"
+      AttributeDefinitions:
+        - AttributeName: "sessionId"
+          AttributeType: "S"
+        - AttributeName: "authorizationCode"
```
Comment (ViveakV) [inline infra-l2-dynamo/template.yaml:71]: Lines 71 to 76 can be deleted

Comment (madankaruppiah) [reply infra-l2-dynamo/template.yaml]: Removed

File: infra-l2-dynamo/template.yaml (hunk 4 above)
Comment (madankaruppiah) [reply infra-l2-dynamo/template.yaml]: Removed

**Review by ViveakV — APPROVED** [review]

**Pull request merged**